src/Node.cpp

```diff
@@ -9,26 +9,37 @@
 
 #include "NodeGraphicsObject.hpp"
 #include "NodeDataModel.hpp"
+#include "NodeStyle.hpp"
 
 #include "ConnectionGraphicsObject.hpp"
 #include "ConnectionState.hpp"
 
 using QtNodes::Node;
 using QtNodes::NodeGeometry;
 using QtNodes::NodeState;
+using QtNodes::NodeStyle;
 using QtNodes::NodeData;
 using QtNodes::NodeDataType;
 using QtNodes::NodeDataModel;
 using QtNodes::NodeGraphicsObject;
 using QtNodes::PortIndex;
 using QtNodes::PortType;
 
+
+static NodeStyle const *
+computeStyle(NodeStyle const *preferredStyle, NodeStyle const &backupStyle)
+{
+  return preferredStyle ? preferredStyle : &backupStyle;
+}
+
+
 Node::
-Node(std::unique_ptr<NodeDataModel> && dataModel)
+Node(std::unique_ptr<NodeDataModel> && dataModel, NodeStyle const& style)
   : _uid(QUuid::createUuid())
   , _nodeDataModel(std::move(dataModel))
   , _nodeState(_nodeDataModel)
-  , _nodeGeometry(_nodeDataModel)
+  , _nodeStyle(computeStyle(_nodeDataModel->nodeStyle(), style))
```

Comment:
And this is maybe a reason to have a pointer.

Reply:
If `computeStyle` returns a `NodeStyle const&`, then `_nodeStyle` could also be a `NodeStyle const&`:

```c++
NodeStyle const&
computeStyle(NodeStyle const *preferredStyle, NodeStyle const &backupStyle)
{
  return preferredStyle ? *preferredStyle : backupStyle;
}
```

I'll change the style members from being `const *` to `const &`, unless you say otherwise